src/assets/scripts/resources-utils.js

```diff
@@ -0,0 +1,282 @@
+}
+
+/*
+ * Extract text from a html string.
+ * @param {String} input - The string to extract text from.
+ * @return The extracted text.
+ */
+function htmlDecode(input) {
+	let el = document.createElement('div');
+	el.innerHTML = input;
+	return el.innerText;
+}
+
+/*
+ * Process each object in the data set to convert some field value to formats for display.
+ * By default sort by publishedYear. 
+ * @param {Array<Object>} dataSet - The data set to process.
+ * @return The same set of the data set with fields converted.
+ */
+function processResourcesDisplayResults(inArray) { // eslint-disable-line no-unused-vars
+	const sortedArray = [];
+	const resultsWithPublishedYear = inArray.filter(result => result.publishedYear);
+	const resultsWithoutPublishedYear = inArray.filter(result => !result.publishedYear);
+	resultsWithPublishedYear.sort((a, b) => {
+		let compare = b.publishedYear.localeCompare(a.publishedYear);
+		if (compare === 0) {
+			compare = a.title.localeCompare(b.title);
+		}
+		return compare;
+	});
+	resultsWithoutPublishedYear.sort((a, b) => (a.title.localeCompare(b.title)));
+
+	sortedArray.push(...resultsWithPublishedYear, ...resultsWithoutPublishedYear);
+
+	return sortedArray.map((oneRecord) => {
+		oneRecord.title = htmlDecode(oneRecord.title);
+		oneRecord.description = stripHtmlTags(oneRecord.description);
+		return oneRecord;
+	});
+}
+
+/*
+ * Render the applied filters
+ * @param {Array} resources - An array of resources to be rendered.
+ * @return directly add the applied filter html to the content selector.
+ */
+function renderAppliedFilters(resources, resourceTopics, resourceTypes) {
+	const appliedFilters = document.querySelectorAll('.filter-checkbox:checked');
+	let appliedFilterHtml = '';
+
+	if (resources.length === 0) {
+		appliedFilterHtml += `<div class="resources-no-results"><p>Sorry, no results were found based on your applied filters.</p></div>`;
+	} else {
+		appliedFilterHtml += `<div class="resources-filtered-number" role="alert"><p>Showing ${resources.length} ${resources.length === 1 ? 'result' : 'results'}</p></div>`
+	}
+
+	appliedFilterHtml += '<h3>Applied filters</h3><div class="filter-tags">';
+
+	for (const appliedFilter of appliedFilters) {
+		const tagType = resourceTopics.find(topicObj => topicObj.value === appliedFilter.id) ? 'topic': 'type';
+		let filterLabel = '';
+		if (tagType === 'topic') {
+			filterLabel = resourceTopics.find(topicObj => topicObj.value === appliedFilter.id).label;
+		} else if (tagType === 'type') {
+			filterLabel = resourceTypes.find(typeObj => typeObj.value === appliedFilter.id).label;
+		}
+
+		const params = new URLSearchParams(window.location.search);
+		params.delete(appliedFilter.name);
+
+		appliedFilterHtml += `<button aria-label="Remove filter resource ${tagType} ${filterLabel}" data-tag=${appliedFilter.id} class="filter-tag" onClick="window.location.href='${window.location.href.split('?')[0]}?${params.toString()}'">
```

Comment:
When removing resource type "code", `aria-label` will generate `Remove filter resource type Code`. Might be clearer to generate "remove the filtering on a resource type Code".